package/libs/xcrypt/Makefile

```diff
@@ -15,16 +15,33 @@ PKG_LICENSE_FILES:=COPYING.LIB
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
 
+PKG_CONFIG_DEPENDS:= \
+	CONFIG_USE_GLIBC \
+	CONFIG_PACKAGE_libcrypt-compat \
+	CONFIG_PACKAGE_libxcrypt
+
 include $(INCLUDE_DIR)/package.mk
 
-define Package/libxcrypt
+define Package/libxcrypt/Default
   SECTION:=libs
   CATEGORY:=Libraries
   URL:=https://github.com/besser82/libxcrypt
   TITLE:=Extended crypt library
+endef
+
+define Package/libxcrypt
+$(Package/libxcrypt/Default)
+  VARIANT:=regular
   BUILDONLY:=1
 endef
 
+define Package/libcrypt-compat
+$(Package/libxcrypt/Default)
+  VARIANT:=compat
+  TITLE+= - libc compatibility
```

Comment:
I guess you mean glibc here

Reply:
I mean "generic libc", not only glibc. This package may be used in future with musl libc too, but this is not a case for now.

Reply:
Currently, the samba4 package will not build due to I think this change.

Reply:
So what needs to be changed to fix?